Unable to compile on Linux

Hey,

I'm very interested in this library and want to try it out for my project.
I develop & want to deploy the project on Linux, but was unable to compile the project (as a dependency & just from the root of this repo).
My setup is using the latest swift version as of now, 6.3.1.
The following errors are thrown:

```
.../ReerJSON/Sources/ReerJSON/JSONDecoderImpl.swift:270:25: error: 'let' binding pattern cannot appear in an expression
 268 |             }
 269 |             return date
 270 |         case .formatted(let formatter):
     |                         `- error: 'let' binding pattern cannot appear in an expression
 271 |             let string = try unboxString(from: value, for: codingPathNode, additionalKey)
 272 |             guard let date = formatter.date(from: string) else {
```
```
.../ReerJSON/Sources/ReerJSON/JSONEncoderImpl.swift:286:25: error: 'let' binding pattern cannot appear in an expression
284 |         case .iso8601:
285 |             return wrapString(_iso8601Formatter.string(from: date))
286 |         case .formatted(let formatter):
    |                         `- error: 'let' binding pattern cannot appear in an expression
287 |             return wrapString(formatter.string(from: date))
288 |         case .custom(let closure):
```

I did some quick digging and found out that the `formatted` strategy is not implemented in the same place as the others. It needs to be provided by the `swift-corelibs-foundation` as this depends on the ObjC runtime on MacOS. But for some reason it's not being picked up as an option when running the switch case
- https://github.com/swiftlang/swift-corelibs-foundation/blob/05825841603d7c403315efc897f07d18d7fb5aa7/Sources/Foundation/JSONEncoder.swift#L13
- https://github.com/swiftlang/swift-foundation/blob/main/Sources/FoundationEssentials/JSON/JSONEncoder.swift#L57-L81

--- a/Sources/ReerJSON/JSONDecoderImpl.swift
+++ b/Sources/ReerJSON/JSONDecoderImpl.swift
@@ -267,6 +267,7 @@ final class JSONDecoderImpl: Decoder {
                 ))
             }
             return date
+        #if !os(Linux)
         case .formatted(let formatter):
             let string = try unboxString(from: value, for: codingPathNode, additionalKey)
             guard let date = formatter.date(from: string) else {
@@ -276,6 +277,7 @@ final class JSONDecoderImpl: Decoder {
                 ))
             }
             return date
+        #endif
         case .custom(let closure):
             return try with(value: value, path: codingPathNode.appending(additionalKey)) {
                 try closure(self)

--- a/Sources/ReerJSON/JSONEncoderImpl.swift
+++ b/Sources/ReerJSON/JSONEncoderImpl.swift
@@ -283,8 +283,10 @@ class JSONEncoderImpl: Encoder {
             return try wrapFloat(1000.0 * date.timeIntervalSince1970, for: additionalKey)
         case .iso8601:
             return wrapString(_iso8601Formatter.string(from: date))
+        #if !os(Linux)
         case .formatted(let formatter):
             return wrapString(formatter.string(from: date))
+        #endif
         case .custom(let closure):
             return try _encodeNestedValue(for: additionalKey) { try closure(date, self) } ?? yyjson_mut_obj(doc)
         @unknown default: fatalError()

